### PR TITLE
Fix [<tailcall>] false positive with yield!

### PR DESCRIPTION
**Pull request by dawedawe**
## Description

Fixes a false positive that @brianrourkeboll came across in #16882 
Thanks @brianrourkeboll for providing the repro code!

## Checklist

- [X] Test cases added
- [ ] Performance benchmarks added in case of performance changes
- [X] Release notes entry updated:
    > Please make sure to add an entry with short succinct description of the change as well as link to this pull request to the respective release notes file, if applicable.
    >
    > Release notes files:
    > - If anything under `src/Compiler` has been changed, please make sure to make an entry in `docs/release-notes/.FSharp.Compiler.Service/<version>.md`, where `<version>` is usually "highest" one, e.g. `42.8.200`
    > - If language feature was added (i.e. `LanguageFeatures.fsi` was changed), please add it to `docs/releae-notes/.Language/preview.md`
    > - If a change to `FSharp.Core` was made, please make sure to edit `docs/release-notes/.FSharp.Core/<version>.md` where version is "highest" one, e.g. `8.0.200`.

    > Information about the release notes entries format can be found in the [documentation](https://fsharp.github.io/fsharp-compiler-docs/release-notes/About.html).
    > Example:
    > * More inlines for Result module. ([PR #16106](https://github.com/dotnet/fsharp/pull/16106))
    > * Correctly handle assembly imports with public key token of 0 length. ([Issue #16359](https://github.com/dotnet/fsharp/issues/16359), [PR #16363](https://github.com/dotnet/fsharp/pull/16363))
    > *`while!` ([Language suggestion #1038](https://github.com/fsharp/fslang-suggestions/issues/1038), [PR #14238](https://github.com/dotnet/fsharp/pull/14238))

    > **If you believe that release notes are not necessary for this PR, please add `NO_RELEASE_NOTES` label to the pull request.**

### CODE REVIEW

File: src/Compiler/Checking/TailCallChecks.fs [hunk 1]
```diff
@@ -222,6 +222,13 @@ and CheckCall cenv args ctxts (tailCall: TailCall) =
             | Expr.App _ -> Some(TailCall.YesFromExpr cenv.g e)
             | IsAppInLambdaBody t -> Some t
             | _ -> None
+        | Expr.App(_funcExpr, _formalType, _typeArgs, args, _range) ->
```
Comment (edgarfgp) [inline src/Compiler/Checking/TailCallChecks.fs:225]: Can we use named arguments here :)

Comment (dawedawe) [reply src/Compiler/Checking/TailCallChecks.fs]: Sure, sorry. Saw that the second when I pushed.

File: tests/FSharp.Compiler.ComponentTests/ErrorMessages/TailCallAttribute.fs [hunk 2]
```diff
@@ -1522,3 +1522,42 @@ namespace N
...
+            match expr with
+            | SynExpr.Sequential(expr1 = expr1; expr2 = SynExpr.Sequential _ as expr2) ->
+                yield dive expr expr.Range (fun expr -> visitor.VisitExpr(path, traverseSynExpr path, (fun _ -> None), expr))
+                let path = SyntaxNode.SynExpr expr :: path
+                yield dive expr1 expr1.Range (traverseSynExpr path)
+                yield! traverseSequentials path expr2   // should not warn
+
+            | _ ->
+                yield dive expr expr.Range (traverseSynExpr path)
+        }
```
Comment (brianrourkeboll) [inline tests/FSharp.Compiler.ComponentTests/ErrorMessages/TailCallAttribute.fs:1558]: Hmm, I wonder whether the check needs to be more specific — maybe only known-safe constructs like `seq` and `async`?

For example, if you change this `seq { … }` to a list `[ … ]` comprehension, it is no longer safe and can blow the stack.

Would the change in this PR warn about the `yield!` if this were a list comprehension?

Comment (dawedawe) [reply tests/FSharp.Compiler.ComponentTests/ErrorMessages/TailCallAttribute.fs:1558]: Oh, thanks. You're correct. I thought we had the list comprehension covered in another test but it's structure is different.
Back to the drawing board...

Comment (dawedawe) [reply tests/FSharp.Compiler.ComponentTests/ErrorMessages/TailCallAttribute.fs:1558]: Alright, we already warn in list/array comprehensions. It's just not in the released compiler, yet.

**Review by psfinaki — APPROVED** [review]
Thanks!

**Review by vzarytovskii — APPROVED** [review]

**Pull request merged**